src/Android/Xamarin.Android/Samples/Augmented Reality/NavigateAR/RoutePlanner.cs

```diff
@@ -0,0 +1,380 @@
+                _stopsOverlay.Graphics.Add(startGraphic);
+
+                // Update help text.
+                _helpLabel.Text = "Tap to set an end point";
+            }
+            else if (_endPoint == null)
+            {
+                // Place the end point.
+                _endPoint = e.Location;
+                Graphic endGraphic = new Graphic(_endPoint, new SimpleMarkerSymbol(SimpleMarkerSymbolStyle.X, System.Drawing.Color.Red, 25));
+                _stopsOverlay.Graphics.Add(endGraphic);
+
+                // Update help text.
+                _helpLabel.Text = "Solving route";
+
+                // Solve the route.
+                SolveRoute();
+            }
+        }
+
+        private void EnableNavigation()
+        {
+            // Pass the route result to the route viewer.
+            RouteViewerAR.PassedRouteResult = _routeResult;
+
+            // Configure the UI.
+            _navigateButton.Click += NavigateButton_Click;
+            _navigateButton.Visibility = Android.Views.ViewStates.Visible;
+            _helpLabel.Text = "You're ready to start navigating!";
+        }
+
+        private void NavigateButton_Click(object sender, System.EventArgs e)
+        {
+            // Start the AR navigation activity.
+            Intent myIntent = new Intent(this, typeof(RouteViewerAR));
+            StartActivity(myIntent);
+        }
+
+        private async void SolveRoute()
+        {
+            try {
+                // Create route parameters and configure them to enable navigation.
+                _routeParameters = await _routeTask.CreateDefaultParametersAsync();
+                _routeParameters.ReturnStops = true;
+                _routeParameters.ReturnDirections = true;
+                _routeParameters.ReturnRoutes = true;
+
+                // Prefer walking directions if available.
+                TravelMode walkingMode = _routeTask.RouteTaskInfo.TravelModes.FirstOrDefault(mode => mode.Name.Contains("Walk")) ?? _routeTask.RouteTaskInfo.TravelModes.First();
+                _routeParameters.TravelMode = walkingMode;
+
+                // Set the stops for the routel
+                Stop stop1 = new Stop(_startPoint);
+                Stop stop2 = new Stop(_endPoint);
+                _routeParameters.SetStops(new[] { stop1, stop2 });
+
+                // Calculate the route.
+                _routeResult = await _routeTask.SolveRouteAsync(_routeParameters);
+
+                // Get the first result.
+                _route = _routeResult.Routes.First();
+
+                // Create and show a graphic for the route.
+                Graphic routeGraphic = new Graphic(_route.RouteGeometry);
+                _routeOverlay.Graphics.Add(routeGraphic);
+
+                // Allow the user to start navigating.
+                EnableNavigation();
+            }
+            catch (Exception ex)
+            {
+                ShowMessage("Failed to calculate route", "Error");
```

Comment:
You can replace the ShowMessage method with
`new AlertDialog.Builder(this).SetMessage("Failed to calculate route").SetTitle("Error").Show();`